### PR TITLE
KAA-1308:Replaced KAA_UNITTESTS_COMPILE with BUILD_TESTING

### PR DESCRIPTION
**Pull request by yverzun**
KAA_UNITTESTS_COMPILE was replased by built-in cmake BUILD_TESTING variable.
Docs and nix configs were updated, because BUILD_TESTING is ON by default.

### CODE REVIEW

File: client/client-multi/client-c/listfiles/UnitTest.cmake [hunk 1]
```diff
@@ -14,12 +14,13 @@
 # limitations under the License.
 #
 
-if(KAA_UNITTESTS_COMPILE)
+include(CTest)
```
Comment (forGGe) [inline client/client-multi/client-c/listfiles/UnitTest.cmake:17]: How about disable testing for all platforms except POSIX?

**Review by forGGe — COMMENTED** [review]
Small proposition added.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Beaglebone/index.md [hunk 2]
```diff
@@ -36,5 +36,5 @@ Now, the toolchain is installed, and it is time to create Kaa application.
 Since BeagleBone (BeagleBone Black) is running Linux, you can refer to [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/#c-sdk-build) for the detailed process of application creation.
 However, remember, you must specify [the CMake C Compiler](http://www.vtk.org/Wiki/CMake_Cross_Compiling#Setting_up_the_system_and_toolchain) when compiling your Kaa application for BeagleBone:
 
-        cmake -DKAA_MAX_LOG_LEVEL=3 -DCMAKE_C_COMPILER="$TOOLCHAIN_PATH/arm-linux-gnueabihf-gcc" ..
+        cmake -DKAA_MAX_LOG_LEVEL=3 -DCMAKE_C_COMPILER="$TOOLCHAIN_PATH/arm-linux-gnueabihf-gcc" -DBUILD_TESTING=off ..
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Beaglebone/index.md]: please, use `OFF` in caps (same for other places)

File: nix/kaa-client-c/default.nix [hunk 3]
```diff
@@ -81,17 +81,17 @@ let
         __propagate:;
       ''
       + target posixSupport "posix"
-              "${lib.optionalString testSupport ''-DCMAKE_BUILD_TYPE=Debug -DKAA_UNITTESTS_COMPILE=on -DKAA_COLLECT_COVERAGE=1''}"
+              "${lib.optionalString testSupport ''-DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=on -DKAA_COLLECT_COVERAGE=1''}"
```
Comment (rasendubi) [inline nix/kaa-client-c/default.nix]: Doesn't work well for `testSupport = false`. In that case, `BUILD_TESTING` is not set and defaults to `ON`. So this won't disable tests.

**Comment by rasendubi** [comment]
You've missed

> (same for other places)

Replace `off` with `OFF` (and `on` -> `ON`) for _all_ occurances.

File: nix/kaa-client-c/default.nix (hunk 3 above)
Comment (rasendubi) [reply nix/kaa-client-c/default.nix]: `${if testSupport ''-DCMAKE_BUILD_TYPE=Debug -DKAA_COLLECT_COVERAGE=1'' else ''-DBUILD_TESTING=OFF''}`

File: nix/kaa-client-c/default.nix [hunk 4]
```diff
@@ -81,17 +81,17 @@ let
         __propagate:;
       ''
       + target posixSupport "posix"
-              "${lib.optionalString testSupport ''-DCMAKE_BUILD_TYPE=Debug -DKAA_UNITTESTS_COMPILE=on -DKAA_COLLECT_COVERAGE=1''}"
+              "${if testSupport ''-DCMAKE_BUILD_TYPE=Debug -DKAA_COLLECT_COVERAGE=1'' else ''-DBUILD_TESTING=OFF''}"
       + target posixSupport "nologs"
-              "${lib.optionalString testSupport ''-DKAA_UNITTESTS_COMPILE=on''} -DKAA_MAX_LOG_LEVEL=0"
+              "${lib.optionalString (!testSupport) ''-DBUILD_TESTING=OFF''} -DKAA_MAX_LOG_LEVEL=0"
```
Comment (rasendubi) [inline nix/kaa-client-c/default.nix]: `if xxx then xxx else xxx`

File: nix/kaa-client-c/default.nix [hunk 5]
```diff
@@ -81,17 +81,17 @@ let
         __propagate:;
       ''
       + target posixSupport "posix"
-              "${lib.optionalString testSupport ''-DCMAKE_BUILD_TYPE=Debug -DKAA_UNITTESTS_COMPILE=on -DKAA_COLLECT_COVERAGE=1''}"
+              "${if testSupport ''-DCMAKE_BUILD_TYPE=Debug -DKAA_COLLECT_COVERAGE=1'' else ''-DBUILD_TESTING=OFF''}"
```
Comment (rasendubi) [inline nix/kaa-client-c/default.nix]: Missed `then`: `if xxx then xxx else xxx`

**Review by rasendubi — APPROVED** [review]

**Pull request merged**